tests/test_scenario_tree.py

```diff
@@ -152,7 +151,37 @@ def test_markov_conditional_probabilities_of_children_large_tree(self):
                 self.assertAlmostEqual(1.0, sum_prob, delta=tol)
 
     def test_markov_read_write_data(self):
-        self.fail("not tested yet")
+        data = {
+            "cost": {
+                "type": "quadratic",
+                "Q": 1,
+                "R": 1
+            },
+            "constraints": {
+                "type": "polyhedral",
+                "x_min": -1,
+                "x_max": 1,
+                "u_min": -1,
+                "u_max": 1
+            },
+            "dynamics": {
+                "type": "linear",
+                "A": 1,
+                "B": 1
+            },
+            "risk": {
+                "type": "A@VR",
+                "alpha": 0.7,
+                "E": 1,
+                "F": 1,
+                "b": 1
+            }
+        }
+        data_5 = ["quadratic", 1, 1, "polyhedral", -1, 1, -1, 1, "linear", 1, 1, "A@VR", 0.7, 1, 1, 1]
```

Comment:
This is very unstructured. I would avoid doing this. Why don't you just pass a dictionary to `set_data_at_node` and store it as such?